### ISSUE TITLE
When container overlay is opened on iOS, VoiceOver users can focus on content behind container overlay

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#2024]
# Bug Report

## Description
When container overlay is opened on iOS, VoiceOver users can focus on content behind container overlay

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-ui/#/components/terra-overlay/overlay/overlay on iOS using VoiceOver
2. Open container based overlay
3. Swipe left or right
4. Focus can be shifted to buttons behind the container overlay

## Expected Behavior
VoiceOver users on iOS cannot focus on elements behind container overlay when it is open.

## Possible Solution
We need a way to mark children in the OverlayContainer with `aria-hidden`. Currently, the `<OverlayContainer />` component just takes children. We could loop through the children, set aria-hidden on all of them which (are not an `Overlay` component or already include aria-hidden attribute). This has complexitity to it as we'd need to keep track of these so when we remove aria-hidden, only the children we originally targeted have the attribute reset.

Another option would be to add a prop to the `OverlayContainer` called `overlay`. This would change the API of `OverlayContainer` where we would expect children to be only the content to render behind the overlay when it is open, and the overlay prop on `OverlayContainer` would take take the `Overlay` component. This would make it a lot easier to split the 2 up and hide the content intended to render behind the overlay when it is open from screen readers.

## Environment

* Component Name and Version:  terra-overlay
* Browser Name and Version: iOS 12 Mobile Safari w/ Voice Over
